src/preset.ts

```diff
@@ -143,8 +144,14 @@ export function evaluateCondition(condition: Condition): boolean {
 }
 
 function evaluateInheritedPresetConditions(preset: Preset, allPresets: Preset[], references: Set<string>): boolean | undefined {
-    const evaluateParent = (parentName: string) => {
+    const evaluateParent = (parentName: string, isChildUserPreset?: boolean) => {
         const parent = getPresetByName(allPresets, parentName);
+        // If the child is not a user preset, the parent should not be a user preset.
+        // eslint-disable-next-line @typescript-eslint/tslint/config
```

Comment:
If you do `if (parent && !isChildUserPreset && parent.isUserPreset)`, you should be able to fix this lint warning.